LLama/ChatSession.cs

```diff
@@ -494,3 +584,50 @@ in OutputTransform
         }
     }
 }
+
+/// <summary>
+/// The state of a chat session in-memory.
+/// </summary>
+public record SessionState
+{
+    /// <summary>
+    /// Saved executor state for the session in JSON format.
+    /// </summary>
+    public string ExecutorState { get; init; }
+
+    /// <summary>
+    /// Saved context state (KV cache) for the session.
+    /// </summary>
+    public State ContextState { get; init; }
+
+    /// <summary>
+    /// The input transform pipeline used in this session.
+    /// </summary>
+    public List<ITextTransform> InputTransformPipeline { get; init; } = new();
+
+    /// <summary>
+    /// The output transform used in this session.
+    /// </summary>
+    public ITextStreamTransform OutputTransform { get; init; } = new LLamaTransforms.EmptyTextOutputStreamTransform();
+
+    /// <summary>
+    /// The history transform used in this session.
+    /// </summary>
+    public IHistoryTransform HistoryTransform { get; init; } = new LLamaTransforms.DefaultHistoryTransform();
+    
+    /// <summary>
+    /// The JSON representation of the chat history for this session.
+    /// </summary>
+    public string History { get; init; } = new ChatHistory().ToJson();
```

Comment:
Similar to the previous comment, I think we should only convert it to string when we need to save the state as file.

Reply:
I think the case here is the same as with the ExecutorState (although I am not 100% positive on that)